base/src/main/java/uk/ac/ebi/atlas/solr/cloud/fullanalytics/SingleCellAnalyticsCollectionProxy.java

```diff
@@ -52,8 +52,14 @@ public SingleCellAnalyticsCollectionProxy(SolrClient solrClient) {
         }
     }
 
-    public QueryResponse query(SolrQueryBuilder<SingleCellAnalyticsCollectionProxy> solrQueryBuilder) {
-        return query(solrQueryBuilder.build());
+    public static String metadataFieldNameToDisplayName(String metadataFieldName) {
+        // The metadata fields are either characteristics or factors
+        if (metadataFieldName.startsWith("characteristic")) {
+            return characteristicFieldNameToDisplayName(metadataFieldName);
+        }
+        else {
+            return factorFieldNameToDisplayName(metadataFieldName);
```

Comment:
Probably both `factorFieldNameToDisplayName` and `characteristicFieldNameToDisplayName` could be refactored to something like `metadataFieldToDisplayName(String fieldName, String prefix)` (where `prefix` would be `factor_`, `characteristic_` or an empty string?

Reply:
Yeah, I think a lot of these methods can be refactored. They're all doing subtly different things in different context and I always forget which one is doing what.